content/en/docs/concepts/configuration/secret.md

```diff
@@ -672,6 +672,16 @@ A container using a Secret as a
 Secret updates.
 {{< /note >}}
 
+{{< feature-state for_k8s_version="v1.18" state="alpha" >}}
+
+For clusters that are extensively using secrets (having at least tens of thousands of unique
+secret to pod mounts), we designed the `Immutable Secrets/ConfigMaps` feature. It allows you
+to mark individual Secrets/ConfigMaps as `Immutable`, which blocks any changes to their data
+and in return:
+- protects you from accidental bad updates that could cause applications outages
+- improves performance of your cluster by significantly reducing load on kube-apiserver, by
+  closing watches for secrets marked as immutable
+
```

Comment:
Added (btw - it's not possible to change the data once mark as immutable - even by priviliged user).

Reply:
> it's not possible to change the data once mark as immutable - even by priviliged user

If I really want to make that change, it sounds like I'd need to delete the cluster and recreate it, then? Or can I delete the Secret and recreate it?

Reply:
:thought_balloon: are there any security implications, for example if a user is able to mark a ConfigMap immutable and thereby break a process that expects to be able to modify it?

Reply:
> Or can I delete the Secret and recreate it?

You can delete the secret and create a new one, but that won't be picked by pods that are mounting it - you will also need to recreate those pods.

> are there any security implications, for example if a user is able to mark a ConfigMap immutable and thereby break a process that expects to be able to modify it?

If a user is able to modify a config map that automation should only modify, it can already break stuff, but setting malicious data in it.

Reply:
> You can delete the secret and create a new one, but that won't be picked by pods that are mounting it - you will also need to recreate those pods.

That's the sort of detail I think readers would be interested in.

Reply:
Added - thanks!